Doc and release notes followup on #26715 (#27431)
zephyrproject-rtos/zephyr: doc/application/index.rst

> The Zephyr :file:`soc/Kconfig` will create the top-level

Reviewer: Remember that `:file:` just prints the argument in a special font. It doesn't actually include the word "file" in the output. So the output here would be "The Zephyr soc/Kconfig will create the top-level", which is probably not what you want. Needs "file" added.

It might be easier just to link to it directly with zephyr_file as well.

```suggestion
The file :zephyr_file:`soc/Kconfig` will create the top-level
```

Reviewer: (I'll trust you to make this update before this PR gets merged or send a follow-up)

``SoC/CPU/Configuration Selection`` menu in Kconfig.

> By providing additional ``SOC_ROOT`` then it is possible to load out-of-tree
> SoCs into this menu.

Reviewer: ```suggestion
Out of tree SoC definitions can be added to this menu using the ``SOC_ROOT``
CMake variable. This variable contains a semicolon-separated list of directories
which contain SoC support files.
```

> Following the structure above, then the following files can be added to load

Reviewer: ```suggestion
Following the structure above, the following files can be added to load
```

more SoCs into the menu.

.. code-block:: none

        soc
        └── arm
            └── st_stm32
                    ├── Kconfig
                    ├── Kconfig.soc

>                     └── Kconfig.defconfig

Reviewer: Can you please emulate the board porting guide by describing these files' contents in detail?

https://docs.zephyrproject.org/latest/guides/porting/board_porting.html#write-kconfig-files

The differences between these files is not clear and IMO should be.

Perhaps the whole thing should be moved to a new "SoC porting guide" and just referenced from here.

Author: I agree, but this has been the case always and is not related to the SOC_ROOT changes.

If acceptable to you, then I would like to do a followup description PR on how to describe out-of-tree soc.
The reason for taking it separately is mostly to get this PR in as fast as possible, so that people become aware of the SOC_ROOT changes.

A dedicated SOC porting guide can then follow, as that is something new.

Reviewer: > I agree, but this has been the case always and is not related to the SOC_ROOT changes.

Are you sure? Didn't this new change add new files?

Author: No, those file where indirectly mentioned as `source` / `osource` in old example:
https://docs.zephyrproject.org/latest/application/index.html#soc-definitions
```
choice
     prompt "SoC/CPU/Configuration selection"

source "$(SOC_DIR)/$(ARCH)/*/Kconfig.soc"

endchoice

menu "Hardware Configuration"
osource "$(SOC_DIR)/$(ARCH)/*/Kconfig"

endmenu
```
as you see, it was presented how to create the menu and then source those files, but without describing their content.

Reviewer: > If acceptable to you, then I would like to do a followup description PR on how to describe out-of-tree soc.

Sounds good, I made https://github.com/zephyrproject-rtos/zephyr/issues/27439 so we don't forget.

> Those Kconfig file may hold description of the SoC, or they can be used to
> load additional SoC Kconfig files.

Reviewer: ```suggestion
The Kconfig files above may describe the SoC or load additional SoC Kconfig files.
```

This will use your custom platform configurations and will generate the
Zephyr binary into your application directory.

> You can also define your project as a Zephyr module and define it as containing
> additional SoCs, :ref:`modules_build_settings`.

Reviewer: ```suggestion
See :ref:`modules_build_settings` for information on setting SOC_ROOT in a module's
:file:`zephyr/module.yml` file.
```
